### PR TITLE
Support top-level enums in CRDs

### PR DESCRIPTION
**Pull request by sbernauer**
Currently only top-level structs are supported. Enums are supported as well (with some limitations) but not at top-level. 

## Motivation
I want to specify a CRD with a top-level enum

## Solution
Added support for top-level enums

### CODE REVIEW

**Comment by sbernauer** [comment]
I'm a little bit surprised, what are the disadvantages that you see when using a top-level enum?

**Comment by teozkr** [comment]
I'd guesstimate that most CRDs will have at least _some_ fields that are common between variants, and using enums will make it harder to transition to those later. You could do something like:

```rust
struct FooKind {
    shared: String,
    #[serde(flatten)]
    method: FooKindMethod,
}

enum FooKindMethod {
    Foo { ... },
    Bar { ... },
}
```

But that feels like it would get clunky and unclear to document which fields are mutually exclusive with which.

**Comment by clux** [comment]
as an addendum to what Teo is saying: 

It's a funny thing about the versioning of kubernetes crds that most people are almost never bumping the apiVersion and instead keep making tiny-non-breaking changes to it (adding new options, merely marking fields as deprecated) until some major release. Enums make that slightly harder. But I don't see any other reasons why you would want to be careful with such a spec.

File: kube-derive/tests/ui/enum_fails.stderr [hunk 1]
```diff
@@ -1,5 +0,0 @@
-error: Enums or Unions can not #[derive(CustomResource)]
- --> $DIR/enum_fails.rs:6:6
-  |
-6 | enum FooSpec {
-  |      ^^^^^^^
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply kube-derive/tests/ui/enum_fails.stderr:5]: I implemented a test for that

**Comment by sbernauer** [comment]
Thanks for your feedback!
I'm with you that the usage of a top-level Enum must thought through but can also think of some cases where it is useful.

**Comment by clux** [comment]
Agreed. Edge-case, but has potential.

CI wise, I think github was struggling earlier, so it looks like you might need to push again or do the DCO thing (the action has details).

**Comment by sbernauer** [comment]
There is something strange going in the clippy check. `cargo +nightly clippy` works for me locally.
I'm trying to look into it.

**Comment by sbernauer** [comment]
Checks are blocked by https://github.com/briansmith/ring/issues/1469

**Comment by clux** [comment]
Oh, ok, whatever. We can fix-forward on clippy. I'l just merge this now that DCO and all important checks are green.

**Comment by sbernauer** [comment]
Thanks a lot!

**Pull request merged**